clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/job/KubernetesRunJobOperation.java

```diff
@@ -0,0 +1,109 @@
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.KubernetesResourcePropertyRegistry;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.job.KubernetesRunJobOperationDescription;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesDeployManifestDescription;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesKind;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.OperationResult;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.op.manifest.KubernetesDeployManifestOperation;
+import com.netflix.spinnaker.clouddriver.kubernetes.v2.security.KubernetesV2Credentials;
+import com.netflix.spinnaker.clouddriver.model.ArtifactProvider;
+import com.netflix.spinnaker.clouddriver.names.NamerRegistry;
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation;
+import com.netflix.spinnaker.moniker.Moniker;
+import com.netflix.spinnaker.moniker.Namer;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+
+@Slf4j
+public class KubernetesRunJobOperation implements AtomicOperation<DeploymentResult> {
+  private static final String OP_NAME = "RUN_KUBERNETES_JOB";
+  private final KubernetesRunJobOperationDescription description;
+  private final KubernetesV2Credentials credentials;
+  private final KubernetesResourcePropertyRegistry registry;
+  private final Namer namer;
+  private final ArtifactProvider provider;
+
+  public KubernetesRunJobOperation(KubernetesRunJobOperationDescription description,
+                                   KubernetesResourcePropertyRegistry registry,
+                                   ArtifactProvider provider) {
+    this.description = description;
+    this.credentials = (KubernetesV2Credentials) description.getCredentials().getCredentials();
+    this.registry = registry;
+    this.provider = provider;
+    this.namer = NamerRegistry.lookup()
+      .withProvider(KubernetesCloudProvider.getID())
+      .withAccount(description.getCredentials().getName())
+      .withResource(KubernetesManifest.class);
+  }
+
+  private static Task getTask() {
+    return TaskRepository.threadLocalTask.get();
+  }
+
+  public DeploymentResult operate(List _unused) {
+    getTask().updateStatus(OP_NAME, "Running Kubernetes job...");
+    KubernetesManifest jobSpec = this.description.getManifest();
+    KubernetesKind kind = jobSpec.getKind();
+    if (kind != KubernetesKind.JOB) {
+      throw new IllegalArgumentException("Only kind of Job is accepted for the V2 Run Job operation.");
+    }
+
+    if (jobSpec.get("metadata") == null) {
+      jobSpec.put("metadata", new HashMap<>());
+    }
+
+    if (!this.description.getNamespace().isEmpty()){
+      jobSpec.setNamespace(this.description.getNamespace());
+    }
+    // append a random string to the job name to avoid collision
```

Comment:
Is this to allow concurrent jobs?  It looks like we're deleting the job after it's done.

Reply:
Correct. Jobs are immutable once created which would prevent you from running them concurrently. I'm open to suggestions on how better to accomplish this, though. This is what the V1 provider does.

Reply:
I think appending the string is fine, no need to change anything.